lib/parse-dir.js

```diff
@@ -1,168 +1,183 @@
-(function() {
-  'use strict';
-
-  var glob = require('glob');
-  var async = require('async');
-  var fs = require('fs');
-  var path = require('path');
-  var noop = function() {};
-
-  // Parsers
-  var jsonParser = require('./parsers/json');
-  var yamlParser = require('./parsers/yaml');
-  var jsParser = require('./parsers/javascript');
-
-  var ParseDir = function() {
-    var that = this;
-
-    var sync = function (pattern) {
-      that.data = [];
-      that.__assync = true;
-
-      var files = glob.sync(pattern);
-
-      if (files) {
-        try {
-          for (var i = 0, len = files.length; i < len; i++) {
-            var file = files[i];
-
-            that.parseFile.call(that, file, noop);
-          }
-          return that.data;
-        } catch (e) {
-          return e;
-        }
-      } else {
-        return [];
+'use strict';
+
+const glob = require('glob');
+const async = require('async');
+const fs = require('fs');
+const path = require('path');
+const noop = function() {};
+
+// Parsers
+const jsonParser = require('./parsers/json');
+const yamlParser = require('./parsers/yaml');
+const jsParser = require('./parsers/javascript');
+
+// used by ParseDir to process and parse individual files:
+class FileProcessor {
+  constructor(options) {
+    this.data = [];
+    this.__async = options.async !== undefined ? options.aync : true;
+  }
+
+  __determineIfDirectory (done, results) {
```

Comment:
I'd actually prefer no underscores at all here.  I'm cool with whatever you guys prefer though.